src/services/suggestionDiagnostics.ts

```diff
@@ -171,16 +171,30 @@ namespace ts {
     }
 
     // should be kept up to date with getTransformationBody in convertToAsyncFunction.ts
-    function isFixablePromiseArgument(arg: Expression): boolean {
+    function isFixablePromiseArgument(arg: Expression, checker: TypeChecker): boolean {
         switch (arg.kind) {
             case SyntaxKind.FunctionDeclaration:
             case SyntaxKind.FunctionExpression:
             case SyntaxKind.ArrowFunction:
                 visitedNestedConvertibleFunctions.set(getKeyFromNode(arg as FunctionLikeDeclaration), true);
                 // falls through
             case SyntaxKind.NullKeyword:
-            case SyntaxKind.Identifier: // identifier includes undefined
                 return true;
+            case SyntaxKind.Identifier:
+            case SyntaxKind.PropertyAccessExpression: {
+                const symbol = checker.getSymbolAtLocation(arg);
+                if (!symbol) {
+                    return false;
+                }
+                if (checker.isUndefinedSymbol(symbol)) {
+                    return true;
+                }
+                const declaration = first(skipAlias(symbol, checker).declarations);
+                if (declaration) {
+                    return isFunctionLikeDeclaration(hasInitializer(declaration) && declaration.initializer ? declaration.initializer : declaration);

```

Comment:
should this maybe be `skipAlias(symbol, checker).declarations.some(d => isFunctionLikeDeclaration(...`?

Reply:
nitpick: this kind of reads backwards to me; it would be easier to read as `isFunctionLike(declaration) || isFunctionLike(hasInitializer(declaration) && declaration.initializer)`, although that last bit doesn't type check I think.

If it's too hard to re-order, don't worry about it.

Reply:
@sandersn done

Reply:
Changed.